parquet-variant/src/main/java/org/apache/parquet/variant/VariantUtil.java

```diff
@@ -0,0 +1,659 @@
+          case DATE:
+            return Variant.Type.DATE;
+          case TIMESTAMP_TZ:
+            return Variant.Type.TIMESTAMP_TZ;
+          case TIMESTAMP_NTZ:
+            return Variant.Type.TIMESTAMP_NTZ;
+          case FLOAT:
+            return Variant.Type.FLOAT;
+          case BINARY:
+            return Variant.Type.BINARY;
+          case LONG_STR:
+            return Variant.Type.STRING;
+          case TIME:
+            return Variant.Type.TIME;
+          case TIMESTAMP_NANOS:
+            return Variant.Type.TIMESTAMP_NANOS;
+          case TIMESTAMP_NANOS_NTZ:
+            return Variant.Type.TIMESTAMP_NANOS_NTZ;
+          case UUID:
+            return Variant.Type.UUID;
+          default:
+            throw new UnsupportedOperationException(
+                String.format("Unknown type in Variant. primitive type: %d", typeInfo));
+        }
+    }
+  }
+
+  private static IllegalArgumentException unexpectedType(Variant.Type type) {
+    return new IllegalArgumentException("Expected type to be " + type);
+  }
+
+  private static IllegalArgumentException unexpectedType(Variant.Type[] types) {
+    return new IllegalArgumentException("Expected type to be one of: " + Arrays.toString(types));
+  }
+
+  static boolean getBoolean(ByteBuffer value) {
+    checkIndex(value.position(), value.limit());
+    int basicType = value.get(value.position()) & BASIC_TYPE_MASK;
+    int typeInfo = (value.get(value.position()) >> BASIC_TYPE_BITS) & PRIMITIVE_TYPE_MASK;
+    if (basicType != PRIMITIVE || (typeInfo != TRUE && typeInfo != FALSE)) {
+      throw unexpectedType(Variant.Type.BOOLEAN);
+    }
+    return typeInfo == TRUE;
+  }
+
+  /**
+   * Returns a long value from Variant value `value[pos...]`.
+   * It is only legal to call it if `getType` returns one of Type.BYTE, SHORT, INT, LONG,
+   * DATE, TIMESTAMP, TIMESTAMP_NTZ, TIME, TIMESTAMP_NANOS, TIMESTAMP_NANOS_NTZ.
+   * If the type is `DATE`, the return value is guaranteed to fit into an int and
+   * represents the number of days from the Unix epoch.
+   * If the type is `TIMESTAMP/TIMESTAMP_NTZ`, the return value represents the number of
+   * microseconds from the Unix epoch.
+   * If the type is `TIME`, the return value represents the number of microseconds since midnight.
+   * If the type is `TIMESTAMP_NANOS/TIMESTAMP_NANOS_NTZ`, the return value represents the number of
+   * nanoseconds from the Unix epoch.
+   * @param value The Variant value
+   * @return The long value
+   */
+  static long getLong(ByteBuffer value) {
+    checkIndex(value.position(), value.limit());
+    int basicType = value.get(value.position()) & BASIC_TYPE_MASK;
+    int typeInfo = (value.get(value.position()) >> BASIC_TYPE_BITS) & PRIMITIVE_TYPE_MASK;
+    if (basicType != PRIMITIVE) {
+      throw new IllegalStateException(
+          "Expect type to be one of: BYTE, SHORT, INT, LONG, TIMESTAMP, TIMESTAMP_NTZ, TIME, TIMESTAMP_NANOS, TIMESTAMP_NANOS_NTZ");
```

Comment:
References to `TIMESTAMP` in this method and javadoc should also include a suffix. We want to avoid any ambiguity because Spark's `timestamp` differs from the SQL standard.